Date new

Rossi-Luciano opened the pull request:
### O que esse PR faz?
Adiciona procedimento para o tratamento de datas (substituirá o anterior).

### Onde a revisão poderia começar?
Por _commits_

### Como este poderia ser testado manualmente?
NA.

### Algum cenário de contexto que queira dar?
NA.

### Screenshots
NA.

### Quais são tickets relevantes?
NA.

### Referências
NA.

Rossi-Luciano committed:
Importação de bibliotecas e padrões

Rossi-Luciano committed:
Adiciona classes para exceções

Rossi-Luciano committed:
Função para padronizar uma data para uma string com dia, mês e ano separados por '-'

Rossi-Luciano committed:
Função para converter um mês, escrito por extenso, em um valor numérico

Rossi-Luciano committed:
Função para fatiar uma data completa em ano, mês e dia

Rossi-Luciano committed:
Função para a padronização de datas no formato ISO YYYY-MM-DD

Rossi-Luciano committed:
Importa bibliotecas

Rossi-Luciano committed:
Adiciona testes

ednilson commented on scielo_scholarly_data/dates.py:
```diff
@@ -0,0 +1,154 @@
+import html
+import unicodedata
+import re
```
@Rossi-Luciano essas bibliotecas não estão sendo usadas.

ednilson reviewed (changes requested):
@Rossi-Luciano precisa tratar as exceções. Acho que pode continuar com base nos testes que fizemos sempre retornando algo, como uma tupla ou dicionario. O importante é informar o resultado (texto normalizado/transformado (ou não)) e se houve exceção, informando qual.

Rossi-Luciano committed:
Remove bibliotecas não utilizadas

rafaelpezzuto commented on scielo_scholarly_data/dates.py:
```diff
@@ -0,0 +1,151 @@
+    if len(text) == 4 and text.isnumeric():
+        return '-'.join([text, month, day])
+    if len(text) == 8 and text.isnumeric():
+        return text[:4] + '-' + text[4:6] + '-' + text[6:]
```
É difícil saber que no meio do texto está o mês e no final o dia.

Porque não usamos aquele procedimento do "date util parser" que já tenta fazer isso por nós?

rafaelpezzuto commented on scielo_scholarly_data/dates.py:
```diff
@@ -0,0 +1,151 @@
+            if '-' in text:
+                break
+            text = text.replace(c, '-')
+        return text
```
O método assume algumas coisas com base no tamanho da string informada e na presença de preposições. Há alguma razão para não tentarmos fazer a padronização usando algo já pronto como o `date util parser`?

@robertatakenaka, não seria melhor fazer um fork do `dateutil.parser` (https://dateutil.readthedocs.io/en/stable/) e adicionar essas melhorias?

rafaelpezzuto commented on scielo_scholarly_data/dates.py:
```diff
@@ -0,0 +1,151 @@
+        Mês numérico.
+    """
+    if month.lower() in MONTHS_DICT.keys():
+        return MONTHS_DICT[month.lower()]
```
O nome do dicionário poderia ser mais explícito. Lendo `MONTHS_DICT` não sei o que é chave nem o que é valor. Eu costumo nomear dicionários usando o padrão `keyname_to_valuename` (às vezes fica comprido, mas facilita a leitura).

MONTHS_DICT seria algo como `text_month_to_numeric_month`

Não sei se @robertatakenaka concorda com isso.

Rossi-Luciano replied:
O nome do dicionário foi atualizado de acordo com a sugestão.

rafaelpezzuto reviewed (commented):
@Rossi-Luciano Fiz algumas observações e gostaria que a @robertatakenaka dessa uma olhada.

No mais, o principal método `convert_to_iso...` está muito mais claro!

ednilson reviewed (commented):
OK

Rossi-Luciano committed:
Corrige parâmetros

Rossi-Luciano committed:
Corrige teste

Rossi-Luciano committed:
Atualiza nome do dicionário

Rossi-Luciano committed:
Adiciona teste para formato 'mês dia ano'

Rossi-Luciano committed:
Renomeia dicionário

Rossi-Luciano committed:
Atualiza nome do dicionário e importa biblioteca

Rossi-Luciano committed:
Padroniza classes de erro

Rossi-Luciano committed:
Refatora standardizes_date

Rossi-Luciano committed:
Atualiza dicionário

Rossi-Luciano committed:
Padroniza erros

Rossi-Luciano committed:
Refatora convert_to_iso_date

ednilson commented on scielo_scholarly_data/dates.py:
```diff
@@ -0,0 +1,151 @@
+    return y, m, d
+
+
+def convert_to_iso_date(text, day='01', month='01', only_year=False):
```
@Rossi-Luciano de acordo com a dica do @rafaelpezzuto e @robertatakenaka vamos usar `underscore` nos métodos auxiliares.

ednilson commented on scielo_scholarly_data/dates.py:
```diff
@@ -0,0 +1,142 @@
+        return TEXT_MONTH_TO_NUMERIC_MONTH[month.lower()]
+
+
+def split_date(text):
```
incluir _ no nome do método.
`def _split_date(text):`

robertatakenaka commented on tests/test_dates.py:
```diff
@@ -129,6 +129,12 @@ def test_convert_to_iso_invalid_format(self):
     def test_convert_to_iso_invalid_full_month(self):
         self.assertRaises(
-            dates.NoneTypeError,
+            dates.InvalidFormatError,
```
Use `InvalidDateValueError`

Rossi-Luciano committed:
Adiciona teste para data na qual o ano é representado por dois dígitos

Rossi-Luciano committed:
Remove importações não utilizadas

Rossi-Luciano committed:
Move TEXT_MONTH_TO_NUMERIC_MONTH para arquivo dates.py

Rossi-Luciano committed:
Adiciona lista de palavras que serão removidas de datas declaradas

Rossi-Luciano committed:
Adiciona exceção

Rossi-Luciano committed:
Renomeia função que é utilizada internamente

Rossi-Luciano committed:
Insere a utilização de WORDS_TO_REMOVE_IN_DATE_STANDARDIZATION para remoção de palavras

Rossi-Luciano committed:
Renomeia função que é utilizada internamente

Rossi-Luciano committed:
Renomeia função que é utilizada internamente

Rossi-Luciano committed:
Atualiza nomes de funções

Rossi-Luciano committed:
Atualiza nomes de funções

Rossi-Luciano committed:
Atualiza exceção

Rossi-Luciano committed:
Ordena TEXT_MONTH_TO_NUMERIC_MONTH

Note que este dicionário será removido quando este novo PR que trata
datas for incorporado

Rossi-Luciano committed:
Adiciona WORDS_TO_REMOVE_IN_DATE_STANDARDIZATION

Note que esta lista foi adiciona e, posteriormente, movida para o
arquivo date.py
Quando este PR for incorporado, esta lista será removida de values.py

ednilson commented on scielo_scholarly_data/dates.py:
```diff
@@ -1,13 +1,5 @@
 from datetime import datetime
-
 from dateutil.parser import parse
```
@Rossi-Luciano este import não está em uso

ednilson commented on scielo_scholarly_data/dates.py:
```diff
@@ -0,0 +1,151 @@
+        Ano (y), mês (m) e dia (d).
+    """
+    try:
+        y, m, d = text.split('-')
```
@rafaelpezzuto e @Rossi-Luciano entendi que esta é uma função 'interna' de uso exclusivo do módulo dates, que  quando chega neste ponto já é previsto que se trata de um 'mês'. Então ok.

Rossi-Luciano committed:
Remove importação não utilizada

rafaelpezzuto reviewed (changes requested):
@Rossi-Luciano Olhei os commits, fiz a instalação e executei os testes.

Também fiz testes manuais simulando o uso da biblioteca. Observei que há um método `convert_to_iso_date` em `scielo_scholarly_data.standardizer` que age diferente do novo método que há em `scielo_scholarly_data.dates`. Acho que nós iremos manter apenas este último, certo?

Veja:
![image](https://user-images.githubusercontent.com/2096125/172670893-81dc3678-0855-4664-aab8-19aaa2ddb984.png)

ednilson commented:
@Rossi-Luciano seria melhor comparar os resultados obtidos do `dates.convert_to_iso_date ` com `standardizer.convert_to_iso_date` para garantir o melhor em dates. E depois poderia remover do `standardizer`

rafaelpezzuto reviewed (approved):
Remover convert_to_iso_date de `core` e substituir o método `core.convert_to_iso_date` por `dates.convert_to_iso_date` em standardizer.

Rossi-Luciano committed:
Remove importações obsoletas

Rossi-Luciano committed:
Remove função obsoleta em core

Rossi-Luciano committed:
Importa convet_to_iso_date de dates

Rossi-Luciano committed:
Remove dicionários obsoletos

Rossi-Luciano committed:
Importa convert_to_iso_date de dates

Rossi-Luciano committed:
Remove testes obsoletos em core

Rossi-Luciano committed:
Remove testes obsoletos para standardizer

Rossi-Luciano committed:
Atualiza version

Rossi-Luciano committed:
Merge branch 'main' into date_new
